src/batch_write.rs

```diff
@@ -59,27 +59,27 @@ impl DbInner {
             let mut guard = self.state.write();
 
             guard.update_clock_tick(now)?;
+            let seq = guard.increment_seq();
```

Comment:
Nit: Seems not ideal (from a perf point of view) to do two back-to-back `state_copy()` calls (first for clock tick, then for seq increment). But perhaps we should not worry about it since it makes the API cleaner.

Reply:
yes, indeed it's likely to be a performance issue, i think this is better to be refactored about this deep copy. however, it might need a bigger footprint to get this refactor implemented, i suppose we can put it in a stand-alone PR. we've already have an issue about this at https://github.com/slatedb/slatedb/issues/371 , i'd like to have a try on this refactor after this seqnum pr 🫡.